Assets/MixedRealityToolkit.Tests/PlayModeTests/SolverTests.cs

```diff
@@ -0,0 +1,246 @@
+        public IEnumerator TestRuntimeInstantiation()
+        {
+            InstantiateTestSolver<Orbital>();
+
+            yield return null;
+        }
+
+        /// <summary>
+        /// Test solver system's ability to change target types at runtime
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestTargetTypes()
+        {
+            Vector3 rightHandPos = Vector3.right * 50.0f;
+            Vector3 customTransformPos = Vector3.up * 50.0f;
+
+            var transformOverride = new GameObject("Override");
+            transformOverride.transform.position = customTransformPos;
+
+            var testObjects = InstantiateTestSolver<Orbital>();
+
+            // Set solver handler to track hands
+            testObjects.handler.TrackedTargetType = TrackedObjectType.HandJoint;
+
+            InputSimulationService inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+
+            // Test orbital around right hand
+            yield return TestHandSolver(testObjects.target, inputSimulationService, rightHandPos, Handedness.Right);
+
+            // Test orbital around head
+            testObjects.handler.TrackedTargetType = TrackedObjectType.Head;
+
+            yield return new WaitForSeconds(SolverUpdateWaitTime);
+
+            Assert.LessOrEqual(Vector3.Distance(testObjects.target.transform.position, Vector3.zero), DistanceThreshold);
+
+            // Test orbital around custom override
+            testObjects.handler.TrackedTargetType = TrackedObjectType.CustomOverride;
+            testObjects.handler.SetTransformOverride(transformOverride.transform);
+
+            yield return new WaitForSeconds(SolverUpdateWaitTime);
+
+            Assert.LessOrEqual(Vector3.Distance(testObjects.target.transform.position, customTransformPos), DistanceThreshold);
+
+            yield return null;
+        }
+
+        /// <summary>
+        /// Tests solver handler's ability to switch hands
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestHandModality()
+        {
+            var testObjects = InstantiateTestSolver<Orbital>();
+
+            // Set solver handler to track hands
+            testObjects.handler.TrackedTargetType = TrackedObjectType.HandJoint;
+
+            // Set and save revelant positions
+            Vector3 rightHandPos = Vector3.right * 20.0f;
+            Vector3 leftHandPos = Vector3.right * -20.0f;
+
+            yield return null;
+
+            InputSimulationService inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+
+            // Test orbital around right hand
+            yield return TestHandSolver(testObjects.target, inputSimulationService, rightHandPos, Handedness.Right);
+
+            // Test orbital around left hand
+            yield return TestHandSolver(testObjects.target, inputSimulationService, leftHandPos, Handedness.Left);
```

Comment:
Would also be interesting have a test that shows what happens when both hands are in the scene - if people start depending on this behavior, we should have a test for that

Reply:
Added to TestHandModality